### PR TITLE
Added admin user

### PR DESCRIPTION
**Pull request by RakeshMaddala**
This PR contains codes for 
1. In registration page, there will be a dropdown to select user type.
2. Introducing admin page which lists all unapproved challenges
3. Introducing the approve button on unapproved challenge listed in admin page.

### CODE REVIEW

**Comment by bharathiraja-subramanian** [comment]
@RakeshMaddala ,
There shouldn't be user selection type in registration page, if we want to provide regular user `ADMIN` role, then it should be in `admin` section of the portal. Please remove this option, it should not be via registration

File: ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java [hunk 1]
```diff
@@ -134,9 +141,36 @@ else if (principal != null) {
         return model;
     }
 
+    @RequestMapping(value="/admin", method = GET)
+    public ModelAndView showAdminPage(Principal principal) {
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java]: Similarly we need to have view for role change for the user

File: ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java [hunk 2]
```diff
@@ -134,9 +141,36 @@ else if (principal != null) {
...
+        UserDetails userInfo = userService.loadUserByUsername(principal.getName());
+        Boolean userIsAdmin = userInfo.getAuthorities().stream().anyMatch(x -> x.getAuthority().equalsIgnoreCase("admin"));
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java]: Use `@RolesAllowed()` instead of checking by code

File: ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java [hunk 3]
```diff
@@ -134,9 +141,36 @@ else if (principal != null) {
...
+        List<Story> unApprovedChallengeDetailList = commonService.findAllByApprovedIsFalse();
+
+        ModelAndView mv = new ModelAndView(ADMIN);
+        mv.addObject(UNAPPROVED_CHALLENGES, unApprovedChallengeDetailList );
+        mv.addObject("userIsAdmin", userIsAdmin);
+        return mv;
+    }
+
+    @RequestMapping(value="/approve/{id}", method = GET)
+    public String approveChallengeEntry(Principal principal, @PathVariable String id) {
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java]: Create separate controller for all admin apis

File: ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java [hunk 4]
```diff
@@ -134,9 +141,36 @@ else if (principal != null) {
...
+    public ModelAndView index(Principal principal) {
+        UserDetails userInfo = userService.loadUserByUsername(principal.getName());
+        Boolean userIsAdmin = userInfo.getAuthorities().stream().anyMatch(x -> x.getAuthority().equalsIgnoreCase("admin"));
+        return new ModelAndView("index", "userIsAdmin", userIsAdmin);
+    }
+
+    @RequestMapping("/home")
+    public ModelAndView home(Principal principal) {
+        UserDetails userInfo = userService.loadUserByUsername(principal.getName());
+        Boolean userIsAdmin = userInfo.getAuthorities().stream().anyMatch(x -> x.getAuthority().equalsIgnoreCase("admin"));
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java]: Remove `isAdmin` checks and use RBAC in UI and Server

File: ideaforce/src/main/java/com/emc/ideaforce/controller/UserDto.java [hunk 5]
```diff
@@ -23,6 +23,7 @@
     @Size(min = 8, max = 20, message = "Password should be 8-20 chars")
     private String password;
     private String matchingPassword;
+    private String[] roles;
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/java/com/emc/ideaforce/controller/UserDto.java]: Remove it..not reqired

File: ideaforce/src/main/java/com/emc/ideaforce/repository/StoryRepository.java [hunk 6]
```diff
@@ -4,6 +4,10 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 public interface StoryRepository extends JpaRepository<Story, String> {
+    List<Story> findAllByApprovedIsFalse();
+    Story findChallengeEntryByIdEquals(String entryId);
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/java/com/emc/ideaforce/repository/StoryRepository.java]: Rename method as `ChallengeEntry` is already renamed

File: ideaforce/src/main/resources/templates/registration.html [hunk 7]
```diff
@@ -36,6 +36,13 @@ <h1 th:text="#{label.regform.title}">form</h1>
         <label th:text="#{label.user.confirmPass}">confirm</label>
         <input type="password" th:field="*{matchingPassword}"/>
     </div>
+    <div>
+        <label>Select Role</label>
+        <select th:field="*{roles}">
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/resources/templates/registration.html]: Shouldnt be here

**Comment by RakeshMaddala** [comment]
@bharathiraja-subramanian - Thanks for your review comments. With recent commit i have addressed all. Please review.

File: ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java [hunk 8]
```diff
@@ -135,8 +139,13 @@ else if (principal != null) {
     }
 
     @GetMapping("/")
-    public String index(Principal principal) {
-        return HOME_VIEW;
+    public ModelAndView index(Principal principal) {
+        return new ModelAndView("index");
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java]: Why is this changed? Where there any issue?

File: ideaforce/src/main/java/com/emc/ideaforce/utils/Utils.java [hunk 9]
```diff
@@ -3,7 +3,7 @@
 public class Utils {
     public static final String PWD_PRIVELEGE = "CHANGE_PASSWORD_PRIVILEGE";
     public static final String REG_USER_ROLE = "ROLE_USER";
-    public static final String ADMIN_ROLE = "ADMIN";
+    public static final String ADMIN_ROLE = "ROLE_ADMIN";
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/java/com/emc/ideaforce/utils/Utils.java]: Is default admin user also having this role?

Comment (RakeshMaddala) [reply ideaforce/src/main/java/com/emc/ideaforce/utils/Utils.java]: yes

File: ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java (hunk 8 above)
Comment (RakeshMaddala) [reply ideaforce/src/main/java/com/emc/ideaforce/controller/CommonController.java]: I was passing isUserAdmin attribute into index. While refactoring i left it as ModelView..
